src/networking.c

```diff
@@ -415,7 +415,8 @@ void _addReplyToBufferOrList(client *c, const char *s, size_t len) {
      * after the command's reply (specifically important during multi-exec). the exception is
      * the SUBSCRIBE command family, which (currently) have a push message instead of a proper reply.
      * The check for executing_client also avoids affecting push messages that are part of eviction. */
-    if (c == server.current_client && (c->flags & CLIENT_PUSHING) &&
+    // TODO: should forbid the ReplyWith* module family api from being called outside the lock?
+    if ((c->flags & CLIENT_PUSHING) && c == server.current_client &&
```

Comment:
start version: 7.2.0
introduced: #12326
reason: `RM_Reply*` is not thread-safe

```
WARNING: ThreadSanitizer: data race (pid=1969717)
  Write of size 8 at 0x558d01162be0 by main thread (mutexes: write M86, write M82, write M84, write M64):
    #0 processCommandAndResetClient /data/redis_fork/src/networking.c:2466 (redis-server+0xe1db4)
    #1 processInputBuffer /data/redis_fork/src/networking.c:2575 (redis-server+0xe1db4)
    #2 readQueryFromClient /data/redis_fork/src/networking.c:2715 (redis-server+0xe28da)
    #3 callHandler /data/redis_fork/src/connhelpers.h:79 (redis-server+0x2943f2)
    #4 connSocketEventHandler /data/redis_fork/src/socket.c:298 (redis-server+0x2943f2)
    #5 aeProcessEvents /data/redis_fork/src/ae.c:436 (redis-server+0x97245)
    #6 aeMain /data/redis_fork/src/ae.c:496 (redis-server+0x97245)
    #7 main /data/redis_fork/src/server.c:7212 (redis-server+0x846d5)

  Previous read of size 8 at 0x558d01162be0 by thread T14:
    #0 _addReplyToBufferOrList /data/redis_fork/src/networking.c:418 (redis-server+0xdc0d6)
    #1 addReplyProto /data/redis_fork/src/networking.c:474 (redis-server+0xdd0a7)
    #2 RM_ReplyWithCallReply /data/redis_fork/src/module.c:3424 (redis-server+0x212732)
    #3 bg_call_worker /data/redis_fork/tests/modules/blockedclient.c:145 (blockedclient.so+0x9c0d)
```

Reply:
they're not related.
this is because main thread many modify `server.current_client` when module thread read it.
however, `c->flags & CLIENT_PUSHING` is always false for module threads, so this is harmless.

Reply:
@sundb this one at the top comment (number 2), says:
> Harm Level: Low

but if that's just an access to a variable and then ignoring what we read from it, isn't that "Harm Level: None"?

Reply:
You are right, it should be `None`.